EmptyStateView component, refactored dashboard layout

kara-o opened the pull request:
## What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
*	Main purpose of this PR was to add the ‘No Donations’ view for both client and donor.  The user story is that client/donor logs in, and once taken to the dashboard ‘home’ screen, if there are no donations available (for client) or no active donations made (for donor), then the screen shows an image with text along the lines of ‘No Donations Available.’
*	This is the view for an ‘empty state’ situation – in this case, the array of donations we received from the db query is empty.  I decided to create a reusable component for this situation.  For any empty state situation, in which there is nothing to show the user, we can use this component which will display the banana logo along with our choice of text:

<img width="412" alt="clientdb" src="https://user-images.githubusercontent.com/44376276/80888720-4f6ea380-8c73-11ea-9643-4e050cef67a6.png">
<img width="418" alt="donordb" src="https://user-images.githubusercontent.com/44376276/80888560-441b7800-8c73-11ea-96c1-5d02eaf57ad5.png">

## Other changes in this PR:
*	Updated background color of DashboardScreen/DonorDashboardScreen to match Figma (will need to decide background of SafeAreaView to match)
*	Removed padding from both dashboard screens as discussed with Tao (global horizontal padding causing issues)
*	Changed where we had Title component, instead of being coupled with Navbar, I separated the two so Navbar is alone, and Title is with the rest of the Dashboard content (in one container).  This made more sense to me based on layout, and I was able to then add padding to just Dashboard content to make it fit with Figma and not affect Header/Navbar.

![Screen Shot 2020-05-02 at 12 41 13 PM](https://user-images.githubusercontent.com/44376276/80888979-61e8dd00-8c73-11ea-8b3e-aad2337de0b5.png)

*	Added DonorDonationScreen to MainStack navigator for functionality of “+” button on DonorDashboardScreen to work

![Screen Shot 2020-05-02 at 12 51 06 PM](https://user-images.githubusercontent.com/44376276/80889931-a4aab500-8c73-11ea-8fde-8700b365a2bf.png)

## Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
I tested out my code with the new NavBar/SafeAreaView changes and they work well with no breaks.

duskcloudxu commented:
I am trying to test the page and wondering how could you see emptyStateView at the donor side since all the donors in the database have some donation already...

kara-o commented:
I temporarily changed the logic on Donations.tsx (line 40)/DonationsOrClaims.tsx (line 41) so it would evaluate to false.

jacksonrya commented on .vscode/settings.json:
```diff
@@ -1,26 +1,35 @@
+			"source.fixAll.eslint": true
+		}
+	},
+	"eslint.codeAction.showDocumentation": {
```
I would remove these empty eslint actions though. Make sure if you're customizing your IDE that you do it in your User/ Global settings and have it override the workspace.

kara-o replied:
Good point!  I reverted this file.

jacksonrya commented on .vscode/launch.json:
```diff
@@ -1,22 +1,22 @@
 {
-    // Use IntelliSense to learn about possible attributes.
```
Probably revert this file. If you have auto-fix linting on save enabled. You can either save without formatting via the VSCode command `File: Save without formatting`, or just don't stage this file.

kara-o replied:
reverted!

jacksonrya commented:
If the image is visible on mobile, then I think this can be merged in as long as a note is provided within file or an issue is created that explains this. I would consider browser fixes low priority.

kara-o commented on src/elements/Paragraph/Paragraph.tsx:
```diff
@@ -3,15 +3,19 @@ import { Text } from 'react-native';
 import styles from './Paragraph.styles';
```
Oh good point!  Will definitely look into that, thanks :)

kara-o commented on src/screens/DashboardScreen/DashboardScreen.styles.ts:
```diff
@@ -7,9 +7,11 @@ const addButtonSize = 110;
 export default StyleSheet.create({
```
Done!

kara-o commented on src/screens/DonorDashboardScreen/DonorDonationScreen/DonorDonationScreen.tsx:
```diff
@@ -20,7 +20,7 @@ import * as colors from '@util/colors';
 import styles from './DonorDonationScreen.styles';
```
done!

kara-o commented on .vscode/launch.json:
```diff
@@ -1,22 +1,22 @@
-            "platform": "ios"
-        }
-    ]
+	// Use IntelliSense to learn about possible attributes.
```
Thanks Tao - I reverted both of those .vscode files

kara-o commented:
Thanks Jackson and Tao for your reviews!  I addressed the comments and will go ahead and merge.  Will make an issue for the browser image issue.  Thanks!
